### PR TITLE
Fix `selectorOrElement` is both `Selector` and `Element`

### PR DESCRIPTION
**Pull request by zamboney**
There are cases (`testcafe` for example) that  the `selectorOrElement` is both `Selector` and `Element`.

This change will make make sure that if `selectorOrElement` is both `Selector` and `Element`, it will choose to use the `Selector` logic instead of the `Element` logic.

|`isSelector`|`isElement`| logic |
|-|-|-|
|false|false| throw an `Error`|
|false|true| `Element` logic|
|true|false| `Selector` logic|
|true|true| `Selector` logic|

### CODE REVIEW

File: js/packages/driver/src/context.ts [hunk 1]
```diff
@@ -319,18 +319,16 @@ export class Context<TDriver, TContext, TElement, TSelector> {
   }
 
   async elements(
-    elementOrSelector: TElement | types.Selector<TSelector>,
+    selectorOrElement: TElement | types.Selector<TSelector>,
```
Comment (kyryloonufriiev) [inline js/packages/driver/src/context.ts:322]: Not important at all, I just like it to be consistent :)
```suggestion
    selectorOrElement: types.Selector<TSelector> | TElement,
```

**Review by kyryloonufriiev — COMMENTED** [review]
The PR looks great as we discussed, nice description! Let's have a dedicated discussion about testing.

File: js/packages/driver/test/unit/element-selector-same.spec.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      driver: mock,
+    })
+    await driver.init()
+  })
+
+  afterEach(async () => {
+    await driver.switchToMainContext()
+  })
+
+  it('should return single element if the `isElement` return `true` and `isSelector` return `false`', async () => {
```
Comment (kyryloonufriiev) [inline js/packages/driver/test/unit/element-selector-same.spec.ts:31]: I would put those tests in `context`, because it is where we testing it

**Review by kyryloonufriiev — CHANGES_REQUESTED** [review]

File: pending-changes.yaml [hunk 3]
```diff
@@ -115,3 +115,9 @@ applitools-for-selenium-ide:
     - 
   bug-fix:
     - Better support in DOM slot element
+"@applitools/driver":
+  feature:
+    - 
+  bug-fix:
+    - |
+      'fix selector issue in `testcafe`, the selector and the element are the same and once there are multi selector the it's only peak the element'
```
[the remark this answers is not in the document]
Comment (zamboney) [reply pending-changes.yaml:123]: np i will remove it

File: js/packages/driver/test/unit/element-selector-same.spec.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    )
+  })
+  it('should throw an error if both `isElement` and `isSelector` return `false`', async () => {
+    const selectorOrElement = {id: selector, notting: true}
+    assert.strictEqual(spec.isElement(selectorOrElement), false, `the selectorOrElement isn't a element`)
+    assert.strictEqual(spec.isSelector(selectorOrElement), false, `the selectorOrElement isn't a selector`)
+    try {
+      await driver.mainContext.elements(selectorOrElement)
+      assert.strictEqual(true, false, 'should thrown error and skip this')
+    } catch (_) {
```
[the remark this answers is not in the document]
Comment (zamboney) [reply js/packages/driver/test/unit/element-selector-same.spec.ts:67]: np i will remove it

File: js/packages/driver/test/unit/element-selector-same.spec.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+      assert.strictEqual(true, true, 'shold thrown erro and get here')
+    }
```
[the remark this answers is not in the document]
Comment (zamboney) [reply js/packages/driver/test/unit/element-selector-same.spec.ts:69]: for the 1st state, `isSelector` and `isElement` return `false`

File: js/packages/driver/test/unit/element-selector-same.spec.ts (hunk 2 above)
Comment (zamboney) [reply js/packages/driver/test/unit/element-selector-same.spec.ts:31]: np

**Review by kyryloonufriiev — APPROVED** [review]
Let's merge it

**Pull request merged**